src/algorithms/optimized_shooting.jl

```diff
@@ -0,0 +1,116 @@
+export periodic_orbit, periodic_orbits, OptimizedShooting
+
+using LeastSquaresOptim: optimize, LevenbergMarquardt
+
+
+"""
+    OptimizedShooting(; kwargs...)
+
+A shooting method [Dednam2014](@cite) combined with Levenberg-Marquardt optimization 
+to find periodic orbits of continuous systems.
+
+## Keyword arguments
+- `Δt::Float64=1e-6`: explicit ODE solver time step. This should correspond to the time step used in the ODE solver specified in the `CoupledODEs` object.
```

Comment:
This is confusing to me. Is this the number you use when you do `step!(ds, Δt)`? If so, this number is unrelated with the solver step.

Reply:
In the paper `Δt` is both the integration step and also the time step used when creating the residual. I have two ideas on how to make this better:
1. say that `Δt` is the time step used to create the residual. Notify user that the original paper used the same value as integration step in their explicit ODE solver
2. remove the `Δt` keyword. Notify user that he should use an explicit solver and specify `dt`. This `dt` is then used when creating the residual instead of `Δt`.